internal/runbits/cves/cves.go

```diff
@@ -39,36 +40,27 @@ func NewCveReport(prime primeable) *CveReport {
 	return &CveReport{prime}
 }
 
-func (c *CveReport) Report(newBuildPlan *buildplan.BuildPlan, oldBuildPlan *buildplan.BuildPlan, names ...string) error {
-	changeset := newBuildPlan.DiffArtifacts(oldBuildPlan, false)
-	if c.shouldSkipReporting(changeset) {
-		logging.Debug("Skipping CVE reporting")
-		return nil
-	}
-
+func (c *CveReport) Report(report *response.ImpactReportResult, names ...string) error {
 	var ingredients []*request.Ingredient
-	for _, artifact := range changeset.Added {
-		for _, ing := range artifact.Ingredients {
-			ingredients = append(ingredients, &request.Ingredient{
-				Namespace: ing.Namespace,
-				Name:      ing.Name,
-				Version:   ing.Version,
-			})
+	for _, i := range report.Ingredients {
+		if i.After == nil {
+			continue // only care about additions or changes
 		}
-	}
 
-	for _, change := range changeset.Updated {
-		if !change.VersionsChanged() {
-			continue // For CVE reporting we only care about ingredient changes
+		if i.Before != nil && i.Before.Version == i.After.Version {
+			continue // only care about changes
 		}
 
-		for _, ing := range change.To.Ingredients {
-			ingredients = append(ingredients, &request.Ingredient{
-				Namespace: ing.Namespace,
-				Name:      ing.Name,
-				Version:   ing.Version,
-			})
-		}
+		ingredients = append(ingredients, &request.Ingredient{
+			Namespace: i.Namespace,
+			Name:      i.Name,
+			Version:   i.After.Version,
+		})
+	}
+
+	if !c.prime.Auth().Authenticated() || len(ingredients) == 0 {
+		logging.Debug("Skipping CVE reporting")
+		return nil
```

Comment:
Can we put the auth check at the top and the `len()` check here? For the len check we don't need a debug entry.

This is pretty nitpicky I realize but it feels wrong to me to have both fire the same debug log, and especially for there not being any ingredients it doesn't seem worth logging at all.